### PR TITLE
Correcting bytes read value with respect to metadata loading

### PR DESCRIPTION
**Pull request by Thejas-bhat**

### CODE REVIEW

File: index/scorch/snapshot_segment.go [hunk 1]
```diff
@@ -35,6 +35,12 @@ type SegmentSnapshot struct {
 	deleted *roaring.Bitmap
 	creator string
 
+	// this flag is needed to identify whether this
+	// segment was mmaped recently, in which case
+	// we consider the loading cost of the metadata
+	// as part of IO stats.
+	mmaped uint64
```
Comment (abhinavdangeti) [inline index/scorch/snapshot_segment.go:42]: Move this to be the first attribute of this struct. Also, a uint32 would suffice?

File: index/scorch/merge.go [hunk 2]
```diff
@@ -429,6 +430,7 @@ type segmentMerge struct {
 	oldNewDocNums map[uint64][]uint64
 	new           segment.Segment
 	notifyCh      chan *mergeTaskIntroStatus
+	mmaped        bool
```
Comment (abhinavdangeti) [inline index/scorch/merge.go:433]: I'd rather we keep the type of this variable as uint for consistency.

**Review by abhinavdangeti — COMMENTED** [review]
Jon should be able to get numbers from your toy build by Thursday.

File: index/scorch/merge.go (hunk 2 above)
Comment (Thejas-bhat) [reply index/scorch/merge.go:433]: Done.

File: index/scorch/snapshot_segment.go (hunk 1 above)
Comment (Thejas-bhat) [reply index/scorch/snapshot_segment.go:42]: yeah it's sufficient, will do.

**Review by abhinavdangeti — APPROVED** [review]
Tentatively approving this.
Let's wait on the results before merging.

**Pull request merged**